scripts/combine/setupCombine.py

```diff
@@ -832,36 +861,44 @@ def outputFolderName(outfolder, card_tool, doStatOnly, postfix):
     return f"{outfolder}/{'_'.join(to_join)}/"
 
 def main(args, xnorm=False):
+    forceNonzero = args.analysisMode == None
+    checkSysts = forceNonzero
+
     fitvar = args.fitvar[0].split("-") if not xnorm else ["count"]
     cardTool = setup(args, args.inputFile[0], fitvar, xnorm)
     cardTool.setOutput(outputFolderName(args.outfolder, cardTool, args.doStatOnly, args.postfix), analysis_label(cardTool))
-    cardTool.writeOutput(args=args, forceNonzero=not args.unfolding, check_systs=not args.unfolding)
+    cardTool.writeOutput(args=args, forceNonzero=forceNonzero, check_systs=checkSysts)
     return
 
 if __name__ == "__main__":
     parser = make_parser()
     args = parser.parse_args()
     
     logger = logging.setup_logger(__file__, args.verbose, args.noColorLogger)
+    
+    isUnfolding = args.analysisMode == "unfolding"
```

Comment:
Is this block duplicated? it also appears in L126-132

Reply:
yeah it is, unfortunately there is no way to propagate all those flags in a simple way, passing them to the function as arguments is also not a good idea

Reply:
I could define them as global variables to set only once, if you prefer, but usually global variables are deprecated

Reply:
No it's ok, fine from my side to merge